IWAID-10 CRUD Pacientes

Chris-THC opened the pull request:
Ticket:
[IWAID-10 CRUD Pacientes](https://iwaids.atlassian.net/browse/IWAID-10?atlOrigin=eyJpIjoiZmVhNTc0ZGY2MmNjNGYwZTg4N2E1M2ZiZTA0NmI2NzAiLCJwIjoiaiJ9)

Overview:
Create the main structure of the table and also create the modal when clicking on edit or add this will behave according to the action chosen. 
Also, create a dummy array to test the behavior and rendering of the components.
If you click on the edit button, the data is added to the form ready to be modified.
And the delete button obtains the data of the selected patient.

What Changed:
I Just add a folder to the original structure, and just add a tag in the patient.jsx file.
I got everything ready to connect to the backend.

How To Test:

Clone the repository
Install the dependencies
npm install
In the project directory, you can run:
npm start
This runs the app in the development mode.
Open http://localhost:3000/ to view it in your browser.


Evidence:

![Strat Table](https://github.com/Chris-THC/imgTemporal/blob/master/newImg/iwaid-10/principal%20v2.png?raw=true)


![Strat Table](https://github.com/Chris-THC/imgTemporal/blob/master/newImg/iwaid-10/buscar%20v2.png?raw=true)


![Add new patient](https://github.com/Chris-THC/imgTemporal/blob/master/newImg/iwaid-10/editar%20v2%20.png?raw=true)


![Update patient](https://github.com/Chris-THC/imgTemporal/blob/master/newImg/iwaid-10/elimina%20v2.png?raw=true)

Chris-THC committed:
I finished iwaid-CRUD-Patients

Chris-THC committed:
I finished iwaid-CRUD-Patients

Chris-THC committed:
Add floating alerts when you click on the add, edit or delete buttons

SendyCruz commented:
Tu rama debe apuntar a la rama donde tienes la estructura inicial y ocupar los archivos que ya definiste

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,177 @@
+          <div className="form-group row">
+            <div className="form-group col-md-8">
+              <label>Fecha de Nacimiento</label>
+              <input
```
Agregar una validación que confirme que la fecha que se está ingresando no es una fecha futura

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,177 @@
+
+          <div className="form-group col-md-4">
+            <label>Ciudad</label>
+            <input
```
Validar que el campo no acepte numeros

SendyCruz commented on src/Components/Patient/TablePatient.jsx:
```diff
@@ -0,0 +1,107 @@
+                  <th>Fecha de Nacimiento</th>
+                  <th>Ciudad</th>
+                  <th>Teléfono</th>
+                  <th>Opcion</th>
```
Opción

SendyCruz commented on src/Components/Routes/Prescriptions.jsx:
```diff
@@ -0,0 +1,10 @@
+  // TODO: The implementation will be worked on the following ticket IWAID-14
+  return (
+    <div>
+      <h1>Prescripciones médicas</h1>
```
Usar singular, me parece que estos cambios pertenecen a otro pr

SendyCruz commented on src/Context/AppContext.jsx:
```diff
@@ -0,0 +1,90 @@
+export const GetTheAppContext = createContext();
+
+export const AppContext = (props) => {
+  const [dataTest, setDataTest] = useState([
```
Agregar un TODO que indique que estos datos duros se van eliminar al hacer la conexión con el back

Chris-THC committed:
Update button styles and add icons

Chris-THC committed:
I added a validation to confirm that the date being entered is not a future date

Chris-THC committed:
Validation not accepting number in city input

Chris-THC committed:
Validation not accepting number in city input

Chris-THC committed:
I edited the words of all the comments that were made in my PR

Chris-THC committed:
I added that array data will be deleted when the frontend connects to the backend

Chris-THC committed:
I added a search filtered by name

Chris-THC commented:
I have addressed all the comments and added the search by name

Chris-THC committed:
I added a select that tells the user what data to search for in the table

SendyCruz commented:
En la sección de filtros esperaria ver los campos inputs disponibles para el filtrado, algo como lo siguiente, tambien agregar un boton para limpiar los filtros
![image](https://github.com/Chris-THC/iwaid-ui/assets/129317505/5e4a9980-e9fd-42dc-ac54-5b73d87872a3)

SendyCruz commented on src/Components/Conteiner.jsx:
```diff
@@ -0,0 +1,36 @@
+    return (
+      <div className=" container alert alert-danger">
+        <h1 className="text-center">
+          Lo sentimos, la página que estás buscando no se encuentra.
```
```suggestion
          Lo sentimos, la página que estás buscando no se encuentra disponible.
```

SendyCruz commented:
esta rama debe apuntar a IWAID-19

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,202 @@
+                  },
+                })}
+              />
+              {errors.fechaNacimineto && (
```
fechaNacimiento

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,202 @@
+          </div>
+        </div>
+
+        <div>
```
El botón debe habilitarse solo si los campos de los datos son correctos

SendyCruz commented on src/Components/Patient/TablePatient.jsx:
```diff
@@ -0,0 +1,159 @@
+                          setActionButtonModal("Editar");
+                        }}
+                      >
+                        <MdChangeCircle className="btn-icon-lg" /> Editar
```
El botón debe habilitarse solo si los campos de los datos son correctos

Chris-THC replied:
Al reutilizar el mismo formulario una vez solucionado el comentario de arriba que era el de agregar, se soluciona el de editar, por ende, el botón se activa siempre y cuando los datos sean válidos.

SendyCruz commented:
Considerar los comentarios para otros CRUDS

Chris-THC committed:
Solving PR comments

Chris-THC committed:
Now the button is activated only when the data is valid

Chris-THC commented:
I have addressed all the comments and I edited the search by name, phone and email. As in the example you showed me  :)

Evidence:

![Strat Table](https://github.com/Chris-THC/imgTemporal/blob/master/crud-pacientes/limpiar.png?raw=true)

Chris-THC committed:
I edited the search form

Chris-THC committed:
I changed the value of the selected in the form

Chris-THC committed:
I added all the indications that you have mentioned, such as validating the form

Chris-THC committed:
I added all the indications that you have mentioned, such as validating the form

Chris-THC committed:
I edited the form the rfc field only accepts capital letters

Chris-THC committed:
I changed only one icon of the edit button

Chris-THC commented:
Here is the evidence of the validation of the form fields 

Evidence: [iwaid-10](https://drive.google.com/file/d/1S33s7JFncCp8BbTevBpml3uqlVyVkP9S/view)

Chris-THC committed:
I changed only one icon of the edit button

Chris-THC committed:
I changed only one icon of the edit button

Chris-THC committed:
I edited the name of the parameters rendered by the tables to the same as in the backend

Chris-THC committed:
Just only added a input to a phone number

SendyCruz commented:
Tendría que apuntar a la 19

Chris-THC commented:
Disculpa creí que ya se había hecho el cambio :(

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,241 @@
+        </div>
+
+        <div className="form-group col-md-8">
+          <label>* RFC</label>
```
El asterisco debe ir del lado derecho y debe ser en color rojo, revisar si se puede ocupar otra etiqueta

SendyCruz commented on src/Components/Patient/MyModalDelete.jsx:
```diff
@@ -0,0 +1,27 @@
+  return (
+    <Modal show={show} onHide={handleClose}>
+      <Modal.Header closeButton>
+        <Modal.Title>Confirmar Eliminación</Modal.Title>
```
Confirmar eliminación

SendyCruz commented on src/Components/Patient/MyModalDelete.jsx:
```diff
@@ -0,0 +1,27 @@
+        <Modal.Title>Confirmar Eliminación</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <p>¿Estás seguro de que deseas eliminar este Paciente?</p>
```
paciente

SendyCruz commented on src/Components/Patient/TablePatient.jsx:
```diff
@@ -0,0 +1,286 @@
+                    </div>
+
+                    <div className="col-md-3 mb-1">
+                      <label>Rango año de nacimiento</label>
```
Por que la fecha de nacimiento sería un rango?

Chris-THC replied:
Por que el ticket solicita un filtro mediante un rango de fecha de nacimiento : https://iwaids.atlassian.net/browse/IWAID-10?atlOrigin=eyJpIjoiMWZjYTQyNzk3NDllNGUwZmFiZmU5ZmU4NjFhMDRmZTYiLCJwIjoiaiJ9

Mostré esa función al equipo de backend y mencionaron que estaba era una buena opción :)

SendyCruz commented on src/Components/Patient/TablePatient.jsx:
```diff
@@ -0,0 +1,286 @@
+              <thead>
+                <tr>
+                  <th>Nombre</th>
+                  <th>Genero</th>
```
Género

SendyCruz commented on src/Components/Patient/TablePatient.jsx:
```diff
@@ -0,0 +1,286 @@
+                  <th>Ciudad</th>
+                  <th>Fecha de nacimiento</th>
+                  <th>RFC</th>
+                  <th>Acción</th>
```
Acciones

SendyCruz commented on src/Components/Patient/TablePatient.jsx:
```diff
@@ -0,0 +1,286 @@
+                        variant="danger"
+                        onClick={() => {
+                          console.log(item.nombre);
+                          // setTextAlert(`Se ha eliminado: ${item.nombre}`);
```
Borrar comentarios

Chris-THC committed:
I solved all comments made in the last PR

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,250 @@
+
+  const handleInputChange = (event) => {
+    const inputValue = event.target.value.toUpperCase();
+    setRfcValue(inputValue); // Actualiza el estado rfcValue con el valor en mayúsculas
```
remover comentarios

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,250 @@
+    setValue("rfc", inputValue); // Actualiza el valor del campo en el formulario en mayúsculas
+    const rfcPattern = /^[A-Z]{4}\d{6}[A-Z\d]{3}$/;
+    if (!rfcPattern.test(inputValue)) {
+      setRfcError("El RFC debe tener un formato válido en mayúsculas.");
```
```suggestion
      setRfcError("El RFC debe tener un formato válido.");
```

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,250 @@
+            />
+            {errors.name && (
+              <span className="text-danger">
+                El campo es solicitado o el campo del nombre no debe de llevar
```
Dato requerido o ...

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,250 @@
+              />
+              {errors.dateOfBirth && (
+                <span className="text-danger">
+                  El dato es solicitado o la fecha no puede ser mayor a la fecha
```
Dato requerido ....

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,250 @@
+              {...register("address", { required: true })}
+            />
+            {errors.address && (
+              <span className="text-danger">El dato es requerido</span>
```
```suggestion
              <span className="text-danger">Dato requerido</span>
```

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,250 @@
+            />
+            {errors.city && (
+              <span className="text-danger">
+                El dato es requerido y no se aceptan números{" "}
```
Dato requerido ...

SendyCruz commented on src/Components/Patient/MyModalDelete.jsx:
```diff
@@ -0,0 +1,27 @@
+        <Modal.Title>Confirmar eliminación</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <p>¿Estás seguro de que deseas eliminar este paciente?</p>
```
```suggestion
        <p>¿Está seguro de que desea eliminar este paciente?</p>
```

Chris-THC committed:
Now you can add, edite, update and delete a Patient

Chris-THC committed:
Update

cgarcia-iwa commented on src/Components/Patient/CrudPatient.jsx:
```diff
@@ -0,0 +1,12 @@
+  const { getDataAllPatients } = useContext(GetTheAppContext);
+  return (
+    <div>
+      <TablePatient dataTable={getDataAllPatients || []} />
```
es posible hacer este cambio?
```suggestion
      <TablePatient dataTable={getAllPatientsData || []} />
```

cgarcia-iwa commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,270 @@
+      setTextAlert(`Paciente ${data.name} actualizado exitosamente`);
+      try {
+        await updatePatientFunction(data, patientId);
+        await getAllPatientDataFunction(setGetDataAllPatients);
```
que indica el setGetData aqui? renombrar a algo que sea mas coherente

cgarcia-iwa commented on src/Context/AppContext.jsx:
```diff
@@ -1,10 +1,76 @@
+    getAllPatientDataFunction(setGetDataAllPatients);
+  }, []);
+
+  // setGetDataAllPatients([
```
si ya no se usa para nada este contenido borrarlo, respaldalo en otro archivo fuera del repositorio si lo necesitas

cgarcia-iwa commented on src/Context/PatientData.js:
```diff
@@ -0,0 +1,56 @@
+export const updatePatientFunction = async (arrayData, idDoctor) => {
+  const urlUpdate = `${doctorURL}${idDoctor}`;
+  try {
+    console.log(urlUpdate);
```
borrar todos los console.log

Chris-THC committed:
This code is only prepared to make the connection to the backend

Chris-THC committed:
This code is only prepared to make the connection to the backend

SendyCruz commented:
en esta sección, esperaría que todos los elementos de filtrado esten en una misma linea o nivel  incluyendo el boton de limpiar filtros
![image](https://github.com/Chris-THC/iwaid-ui/assets/129317505/0ff2da98-c84e-47a9-ad04-4262853c9fa2)

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,269 @@
+        await getAllPatientDataFunction(setGetAllPatientsData);
+        handleShowFloatAlter();
+      } catch (error) {
+        setTextAlert("Error al agregar al paciente");
```
```suggestion
        setTextAlert("Error al actualizar al paciente");
```

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,269 @@
+            />
+            {errors.name && (
+              <span className="text-danger">
+                Dato requerido o el campo del nombre no debe de llevar números
```
```suggestion
                Dato requerido o solo acepta letras
```

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,269 @@
+                <option value="Otro">Otro</option>
+              </select>
+              {errors.gender && (
+                <span className="text-danger">El dato es requerido</span>
```
```suggestion
                <span className="text-danger">Dato es requerido</span>
```

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,269 @@
+            />
+            {errors.phoneNumber?.type === "required" && (
+              <span className="text-danger">
+                El número de teléfono es requerido
```
```suggestion
                Dato requerido
```

SendyCruz commented on src/Components/Patient/MyModalDelete.jsx:
```diff
@@ -0,0 +1,52 @@
+              await deletePatientFunction(patientId);
+              await getAllPatientDataFunction(setGetDataAllPatients);
+              handleButtonClick();
+              setTextAlert(`Se eliminó al médico ${dataUserPatient.name}`);
```
```suggestion
              setTextAlert(`Se eliminó al paciente ${dataUserPatient.name}`);
```

SendyCruz commented on src/Components/Patient/MyModalDelete.jsx:
```diff
@@ -0,0 +1,52 @@
+              setTextAlert(`Se eliminó al médico ${dataUserPatient.name}`);
+              handleShowFloatAlter();
+            } catch (error) {
+              setTextAlert("Error al eliminar el médico");
```
```suggestion
              setTextAlert("Error al eliminar el paciente");
```

SendyCruz commented on src/Components/Patient/TablePatient.jsx:
```diff
@@ -0,0 +1,315 @@
+                          setSearchByCity("");
+                          setSearchByRfc("");
+                        }}
+                        placeholder="Buscar por año de nacimiento..."
```
```suggestion
                        placeholder="Buscar por fecha de nacimiento..."
```

SendyCruz commented on src/Components/Routes/Patient.jsx:
```diff
@@ -1,10 +1,20 @@
 export const Patient = () => {
+  const { showFloatingAlert, handleCloseFloatAlert, textAlert } =
+    useContext(GetTheAppContext);
   // TODO: The implementation will be worked on the following ticket IWAID-10
```
remover

Chris-THC committed:
I attended all iwaid-10's PR comments

Chris-THC committed:
I attended all iwaid-10's PR comments

Chris-THC committed:
I attended all iwaid-10's PR comments

SendyCruz commented:
Actualizar rama con develop

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,273 @@
+                <option value="Otro">Otro</option>
+              </select>
+              {errors.gender && (
+                <span className="text-danger">Dato es requerido</span>
```
```suggestion
                <span className="text-danger">Dato requerido</span>
```

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,273 @@
+              </span>
+            )}
+            {errors.phoneNumber?.type === "maxLength" && (
+              <span className="text-danger">Debe tener 10 dígitos</span>
```
```suggestion
              <span className="text-danger">Máximo 10 dígitos</span>
```

SendyCruz commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,273 @@
+              type="email"
+              className="form-control"
+              autoComplete="off"
+              placeholder="Agregar su email"
```
```suggestion
              placeholder="example@ex.com"
```

SendyCruz commented on src/Context/PatientData.js:
```diff
@@ -0,0 +1,47 @@
+import axios from "axios";
+
+const doctorURL = "http://localhost:8081/iwaid/patients/";
```
```suggestion
const patientURL = "http://localhost:8081/iwaid/patients/";
```

SendyCruz commented on src/Context/PatientData.js:
```diff
@@ -0,0 +1,47 @@
+
+export const getAllPatientDataFunction = async (setGetAllPatients) => {
+  try {
+    const response = await axios.get(doctorURL);
```
```suggestion
    const response = await axios.get(patientURL);
```

SendyCruz commented on src/Context/PatientData.js:
```diff
@@ -0,0 +1,47 @@
+};
+
+export const updatePatientFunction = async (arrayData, idDoctor) => {
+  const urlUpdate = `${doctorURL}${idDoctor}`;
```
```suggestion
  const urlUpdate = `${patientURL}${idPatient}`;
```

SendyCruz commented on src/Context/PatientData.js:
```diff
@@ -0,0 +1,47 @@
+  }
+};
+
+export const deletePatientFunction = async (idDoctor) => {
```
```suggestion
export const deletePatientFunction = async (idPatient) => {
```

SendyCruz commented on src/Context/PatientData.js:
```diff
@@ -0,0 +1,47 @@
+export const deletePatientFunction = async (idDoctor) => {
+  console.log(idDoctor);
+  try {
+    const urlDelete = `${doctorURL}${idDoctor}`;
```
```suggestion
    const urlDelete = `${patientURL}${idPatient}`;
```

SendyCruz commented on src/Context/PatientData.js:
```diff
@@ -0,0 +1,47 @@
+
+export const createPatientFunction = async (arrayData) => {
+  try {
+    const response = await axios.post(doctorURL, arrayData);
```
```suggestion
    const response = await axios.post(patientURL, arrayData);
```

SendyCruz commented on src/Context/PatientData.js:
```diff
@@ -0,0 +1,47 @@
+  }
+};
+
+export const updatePatientFunction = async (arrayData, idDoctor) => {
```
```suggestion
export const updatePatientFunction = async (arrayData, idPatient) => {
```

Chris-THC committed:
The modifications indicated in the comments of the PR have been made

SendyCruz committed:
Merge remote-tracking branch 'origin/develop' into iwaid-10-crud-pacientes

# Conflicts:
#	src/Context/AppContext.jsx

Chris-THC committed:
Now you can create, reade, update and delete a patient

cgarcia-iwa commented on src/Components/Patient/FormPatient.jsx:
```diff
@@ -0,0 +1,273 @@
+            />
+            {errors.name && (
+              <span className="text-danger">
+                Dato requerido o solo acepta letras
```
considerar en el futuro separar los errores como en otro crud

Chris-THC committed:
Now you can create, reade, update and delete a patient
